orb-ui/src/sound/mod.rs

```diff
@@ -408,13 +496,59 @@ impl Player for Fake {
             tracing::error!("Sound file not found: {:?}", sound_type);
         }
     }
+
+    /// Finds the sound file path for the given sound type and sends it to the
+    /// sound player **only if the sink is empty**
+    fn try_queue(&self, sound_type: Type) -> Result<bool> {
+        if let Ok(sink) = self.sink.lock() {
+            if sink.empty() {
+                if let Some(sound_file) = self.sound_files.get(&sound_type) {
+                    if let Some(sound_file) = sound_file.value() {
+                        if let Ok(mut tx_queue) = self.queue_file.lock() {
+                            tx_queue
+                                .try_send(sound_file.clone())
+                                .wrap_err("Failed to queue sound")?;
+                            return Ok(true);
+                        }
+                    } else {
+                        tracing::error!(
+                            "Sound file {:?} doesn't have a known file path",
+                            sound_type
+                        );
+                    }
+                } else {
+                    tracing::error!("Sound file not found: {:?}", sound_type);
+                }
+            }
+        }
+        Ok(false)
+    }
```

Comment:
same thing here, almost all of this can be collapsed such that its not a bunch of nested ifs.